compiler/z/codegen/SystemLinkage.cpp

```diff
@@ -931,10 +932,21 @@ TR::S390zLinuxSystemLinkage::S390zLinuxSystemLinkage(TR::CodeGenerator * codeGen
    setReturnAddressRegister (TR::RealRegister::GPR14 );
 
    setGOTPointerRegister   (TR::RealRegister::GPR12  );
+
+#if defined(OMRZTPF)
+/*
+ * z/TPF os is a 64 bit target
+ */
+   setOffsetToRegSaveArea     (16);    // z/TPF is a 64 bit target only
+   setGPRSaveAreaBeginOffset  (16);    //x'10'  see ICST_R2 in tpf/icstk.h
+   setGPRSaveAreaEndOffset    (160);   //x'a0'  see ICST_PAT in tpf/icstk.h
+   setOffsetToFirstParm       (448);   //x'1c0' see ICST_PAR in tpf/icstk.h
+#else
```

Comment:
@fjeremic just in case you weren't aware... z/TPF needs to push at least 448 byte stack frames.  Anything smaller then that will corrupt our linkage code.